crates/bevy_reflect/src/impls/glam.rs

```diff
@@ -1,44 +1,190 @@
+    Constructor(Default::default())
+    BevyReflectPath(self::bevy_reflect)
+    #[reflect(PartialEq, Serialize, Deserialize)]
+    struct DVec2 {
+        x: f64,
+        y: f64
+    }
+);
+impl_reflect_struct_and_from_reflect_struct!(
+    Constructor(Default::default())
+    BevyReflectPath(self::bevy_reflect)
+    #[reflect(PartialEq, Serialize, Deserialize)]
+    struct DVec3 {
+        x: f64,
+        y: f64,
+        z: f64
+    }
+);
+impl_reflect_struct_and_from_reflect_struct!(
+    Constructor(Default::default())
+    BevyReflectPath(self::bevy_reflect)
+    #[reflect(PartialEq, Serialize, Deserialize)]
+    struct DVec4 {
+        x: f64,
+        y: f64,
+        z: f64,
+        w: f64
+    }
+);
+
+impl_reflect_struct_and_from_reflect_struct!(
+    Constructor(Default::default())
+    BevyReflectPath(self::bevy_reflect)
+    #[reflect(PartialEq, Serialize, Deserialize)]
+    struct Mat3 {
+        x_axis: Vec3,
+        y_axis: Vec3,
+        z_axis: Vec3
+    }
+);
+impl_reflect_struct_and_from_reflect_struct!(
+    Constructor(Default::default())
+    BevyReflectPath(self::bevy_reflect)
+    #[reflect(PartialEq, Serialize, Deserialize)]
+    struct Mat4 {
+        x_axis: Vec4,
+        y_axis: Vec4,
+        z_axis: Vec4,
+        w_axis: Vec4
+    }
+);
+
+impl_reflect_struct_and_from_reflect_struct!(
+    Constructor(Default::default())
+    BevyReflectPath(self::bevy_reflect)
+    #[reflect(PartialEq, Serialize, Deserialize)]
+    struct DMat3 {
+        x_axis: DVec3,
+        y_axis: DVec3,
+        z_axis: DVec3
+    }
+);
+impl_reflect_struct_and_from_reflect_struct!(
+    Constructor(Default::default())
+    BevyReflectPath(self::bevy_reflect)
+    #[reflect(PartialEq, Serialize, Deserialize)]
+    struct DMat4 {
+        x_axis: DVec4,
+        y_axis: DVec4,
+        z_axis: DVec4,
+        w_axis: DVec4
+    }
+);
+
+// Quat fields are read-only (as of now), and reflection is currently missing
+// mechanisms for read-only fields. I doubt those mechanisms would be added,
+// so for now quaternions will remain as values. They are represented identically
+// to Vec4 and DVec4, so you may use those instead and convert between.
 impl_reflect_value!(Quat(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DMat3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DMat4(PartialEq, Serialize, Deserialize));
 impl_reflect_value!(DQuat(PartialEq, Serialize, Deserialize));
-
-impl_from_reflect_value!(IVec2);
-impl_from_reflect_value!(IVec3);
-impl_from_reflect_value!(IVec4);
-impl_from_reflect_value!(UVec2);
-impl_from_reflect_value!(UVec3);
-impl_from_reflect_value!(UVec4);
-impl_from_reflect_value!(Vec2);
-impl_from_reflect_value!(Vec3);
-impl_from_reflect_value!(Vec4);
-impl_from_reflect_value!(Vec3A);
-impl_from_reflect_value!(DVec2);
-impl_from_reflect_value!(DVec3);
-impl_from_reflect_value!(DVec4);
-impl_from_reflect_value!(Mat3);
-impl_from_reflect_value!(Mat4);
-impl_from_reflect_value!(Quat);
```

Comment:
If Quat is to be left as a value then this macro call and the one for DQuat should remain.

Reply:
True, fixed